src/nvim/CMakeLists.txt

```diff
@@ -186,6 +186,10 @@ if(NOT MSVC)
     set_source_files_properties(
       eval/funcs.c PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-conversion")
   endif()
+  # lua-cjson: ignore strict and missing prototypes for cjson
+  file(GLOB CJSON_SOURCES ../cjson/*.c)
+  set_source_files_properties(
+    ${CJSON_SOURCES} PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-strict-prototypes -Wno-missing-prototypes -Wno-conversion")
```

Comment:
"missing prototypes" means there is a build ordering issue or the headers are missing. shouldn't ignore it

Reply:
I fixed the missing prototypes warning, (it was from unused functions), are we ok with silencing strict prototypes?

Reply:
For context, the remaining warnings are from strict prototypes in the floating point conversion files, which I did not modify: 
```
[311/334] Building C object src/nvim/CMakeFiles/nvim.dir/__/cjson/fpconv.c.o
In file included from ../src/cjson/fpconv.c:36:
../src/cjson/fpconv.h:15:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
   15 | extern void fpconv_init();
      | ^~~~~~
../src/cjson/fpconv.c:58:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
   58 | static void fpconv_update_locale()
      |             ^~~~~~~~~~~~~~~~~~~~
../src/cjson/fpconv.c:205:6: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  205 | void fpconv_init()
      |      ^~~~~~~~~~~
[315/334] Building C object src/nvim/CMakeFiles/nvim.dir/__/cjson/lua_cjson.c.o
In file included from ../src/cjson/lua_cjson.c:51:
../src/cjson/fpconv.h:15:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
   15 | extern void fpconv_init();
      | ^~~~~~
[334/334] Generating doc/tags
```

Reply:
For reference, clang is complaining about two different warnings
> clang: No previous prototype for function 'fpconv_init' [-Wmissing-prototypes]
clang: This old-style function definition is not preceded by a prototype [-Wstrict-prototypes]

Reply:
I know what the warnings are, and I already resolved the former.